packages/rpc-methods/src/restricted/getBip32Entropy.ts

```diff
@@ -0,0 +1,260 @@
+import { BIP32Node, JsonSLIP10Node, SLIP10Node } from '@metamask/key-tree';
+
+import { SnapCaveatType } from '../caveats';
+import { isEqual } from '../utils';
+
+const INDEX_REGEX = /^\d+'?$/u;
+
+const targetKey = 'snap_getBip32Entropy';
+
+export type GetBip32EntropyMethodHooks = {
+  /**
+   * @returns The mnemonic of the user's primary keyring.
+   */
+  getMnemonic: () => Promise<string>;
+
+  /**
+   * Waits for the extension to be unlocked.
+   *
+   * @returns A promise that resolves once the extension is unlocked.
+   */
+  getUnlockPromise: (shouldShowUnlockRequest: boolean) => Promise<void>;
+};
+
+type GetBip32EntropySpecificationBuilderOptions = {
+  allowedCaveats?: Readonly<NonEmptyArray<string>> | null;
+  methodHooks: GetBip32EntropyMethodHooks;
+};
+
+type GetBip32EntropySpecification = ValidPermissionSpecification<{
+  permissionType: PermissionType.RestrictedMethod;
+  targetKey: typeof targetKey;
+  methodImplementation: ReturnType<typeof getBip32EntropyImplementation>;
+  allowedCaveats: Readonly<NonEmptyArray<string>> | null;
+  validator: PermissionValidatorConstraint;
+}>;
+
+type GetBip32EntropyParameters = {
+  path: (`${number}` | `${number}'`)[];
+  curve?: 'secp256k1' | 'ed25519';
+};
+
+/**
+ * Validate a caveat path object. The object must consist of a `path` array and
+ * optionally a `curve` string. Paths must start with `m`, and must contain at
+ * least two indices. If `ed25519` is used, this checks if all the path indices
+ * are hardened.
+ *
+ * @param value - The value to validate.
+ * @throws If the value is invalid.
+ */
+function validatePath(
+  value: unknown,
+): asserts value is GetBip32EntropyParameters {
+  // TODO: Use `@metamask/utils`. The extension currently uses an old version of
+  // `utils`, so this is just a workaround for testing.
+  const isPlainObject = (v: unknown): v is Record<string, unknown> => {
+    return true;
+  };
+
+  if (!isPlainObject(value)) {
+    throw ethErrors.rpc.invalidParams({
+      message: 'Expected a plain object.',
+    });
+  }
+
+  if (
+    !('path' in value) ||
+    !Array.isArray(value.path) ||
+    value.path.length === 0
+  ) {
+    throw ethErrors.rpc.invalidParams({
+      message: `Invalid "path" parameter. The path must be a non-empty BIP-32 derivation path array.`,
+    });
+  }
+
+  if (value.path[0] !== 'm') {
+    throw ethErrors.rpc.invalidParams({
+      message: `Invalid "path" parameter. The path must start with "m".`,
+    });
+  }
+
+  if (
+    value.path
+      .slice(1)
+      .some((v) => typeof v !== 'string' || !INDEX_REGEX.test(v))
+  ) {
+    throw ethErrors.rpc.invalidParams({
+      message: `Invalid "path" parameter. The path must be a valid BIP-32 derivation path array.`,
+    });
+  }
+
+  if (value.path.length < 3) {
+    throw ethErrors.rpc.invalidParams({
+      message: `Invalid "path" parameter. Paths must have at least two indices.`,
+    });
+  }
+
+  if (
+    !('curve' in value) ||
```

Comment:
We're about to outlaw the `in` operator via the lint config, so we should refactor all uses of it in this PR to use `hasProperty` from `@metamask/utils`.

Reply:
I wasn't aware of this function. Will change it!